server/queue/queue-jms/src/test/java/org/apache/james/queue/jms/AbstractJMSMailQueueTest.java

```diff
@@ -114,6 +114,19 @@ public void testFIFO() throws Exception {
         assertEquals(0, queue.getSize());
     }
 
+    @Test
+    public void dequeueShouldPreserveState() throws Exception {
+        Mail mail = createMail();
+        String state = "state";
+        mail.setState(state);
+
+        queue.enQueue(mail);
+
+        MailQueueItem item = queue.deQueue();
+        assertEquals(state, item.getMail().getState());
```

Comment:
why no assertj ?

Reply:
Yes, but I won't refactor the entire test class....

Reply:
and why not ? there's about 50 of it

Reply:
I wrote only this test using assertJ